plugins/analysis-nori/src/main/java/org/elasticsearch/index/analysis/NoriTokenizerFactory.java

```diff
@@ -48,32 +50,24 @@ public NoriTokenizerFactory(IndexSettings indexSettings, Environment env, String
     }
 
     public static UserDictionary getUserDictionary(Environment env, Settings settings) {
-        if (settings.get(USER_DICT_PATH_OPTION) != null && settings.get(USER_DICT_RULES_OPTION) != null) {
-            throw new ElasticsearchException("It is not allowed to use [" + USER_DICT_PATH_OPTION + "] in conjunction" +
-                " with [" + USER_DICT_RULES_OPTION + "]");
-
+        List<String> ruleList = Analysis.getWordList(env, settings, USER_DICT_PATH_OPTION, USER_DICT_RULES_OPTION);
+        StringBuilder sb = new StringBuilder();
+        if (ruleList == null || ruleList.isEmpty()) {
+            return null;
         }
-        String path = settings.get(USER_DICT_PATH_OPTION);
-        if (path != null) {
-            try (Reader rulesReader = Analysis.getReaderFromFile(env, settings, USER_DICT_PATH_OPTION)) {
-                return rulesReader == null ? null : UserDictionary.open(rulesReader);
-            } catch (IOException e) {
-                throw new ElasticsearchException("failed to load nori user dictionary", e);
-            }
-        } else {
-            List<String> rulesList = settings.getAsList(USER_DICT_RULES_OPTION, Collections.emptyList(), false);
-            if (rulesList == null || rulesList.size() == 0) {
-                return null;
-            }
-            StringBuilder sb = new StringBuilder();
-            for (String line : rulesList) {
-                sb.append(line).append(System.lineSeparator());
-            }
-            try (Reader rulesReader = new StringReader(sb.toString())) {
-                return UserDictionary.open(rulesReader);
-            } catch (IOException e) {
-                throw new ElasticsearchException("failed to load nori user dictionary", e);
+        // check for duplicate terms
+        Set<String> terms = new HashSet<>();
+        for (String line : ruleList) {
+            String[] split = line.split("\\s+");
+            if (terms.add(split[0]) == false) {
+                throw new IllegalArgumentException("Found duplicate term: [" + split[0] + "] in user dictionary. ");
```

Comment:
If this stays an error or maybe gets converted to a log message, the line number would be a helpful debugging information for the user the file.

Reply:
Agreed, I'll make sure we report the line number in the follow up pr.